go/tools/extract_package/extract_test.go

```diff
@@ -0,0 +1,21 @@
+package main
+
+import (
+	"testing"
+)
+
+func TestExtract(t *testing.T) {
+	for _, fname := range []string{
+		"extract.go",
+		"extract_test.go",
+		"ignored.go",
+	} {
+		name, err := extract(fname)
```

Comment:
this is a bit iffy, because it depends on the cwd in which the test executes. This should probably use runfiles if it is to run in bazel. You can work around this by inlining the input source, and writing it to a ioutil.TempFile and running on that.